ogre/src/OgreMesh.cc

```diff
@@ -58,6 +58,8 @@ void OgreMesh::Destroy()
 
   auto ogreScene = std::dynamic_pointer_cast<OgreScene>(this->Scene());
 
+  Ogre::MeshManager::getSingleton().remove(
```

Comment:
this removes the template mesh. Should we have a similar logic as what you did for textures and check reference count to the mesh first before removing it? Same for the `remove` call in `OgreSubMesh::Destroy()`